addons/account/wizard/account_move_send.py

```diff
@@ -227,6 +245,13 @@ def _get_mail_template_attachments_data(self, mail_template):
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
+    @api.depends('move_ids')
+    def _compute_show_pdf_template_menu(self):
+        available_templates_count = self.env['ir.actions.report'].search_count([('is_invoice_report', '=', True)])
+        for wizard in self:
+            # show pdf template menu if there are more than 1 template available and there is at least one move that needs a pdf
+            wizard.show_pdf_template_menu = False if available_templates_count == 1 else any(self._need_invoice_document(move) for move in wizard.move_ids)
```

Comment:
Could we not do something like this (haven't tested if the logic is preserved) 
==> available_templates_count != 1 and any(self._need_invoice_document(move) for move in wizard.move_ids)